packages/core/src/components/Table/Table/__stories__/Table.stories.tsx

```diff
@@ -402,14 +403,14 @@ export const Borders = {
 export const TableHeaderFunctionality = {
   render: () => {
     const [tableData, setTableData] = useState(emailTableData);
-    const [sorting, setSorting] = useState({});
+    const [sorting, setSorting] = useState<{[key: string]: "asc" | "desc" | "none"}>({});
 
-    const onSort = (columnId, sortState) => {
+    const onSort = (columnId: string, sortState: "asc" | "desc" | "none") => {
       setSorting({
         [columnId]: sortState
       });
 
-      setTableData(sort(columnId, sortState, tableData));
+      setTableData(sort(columnId as keyof typeof tableData[number], sortState, tableData));
```

Comment:
why not to use the generic type? It was added to `sort` in this PR

Reply:
In this case generic type is automatically determined from the provided `tableData` object